core/txpool/blobpool/blobpool.go

```diff
@@ -0,0 +1,1460 @@
+		next    = p.state.GetNonce(from)
+	)
+	if uint64(len(p.index[from])) > tx.Nonce()-next {
+		// Account can support the replacement, but the price bump must also be met
+		prev := p.index[from][int(tx.Nonce()-next)]
+		switch {
+		case tx.GasFeeCapIntCmp(prev.execFeeCap.ToBig()) <= 0:
+			return fmt.Errorf("%w: new tx gas fee cap %v <= %v queued", txpool.ErrReplaceUnderpriced, tx.GasFeeCap(), prev.execFeeCap)
+		case tx.GasTipCapIntCmp(prev.execTipCap.ToBig()) <= 0:
+			return fmt.Errorf("%w: new tx gas tip cap %v <= %v queued", txpool.ErrReplaceUnderpriced, tx.GasTipCap(), prev.execTipCap)
+		case tx.BlobGasFeeCapIntCmp(prev.blobFeeCap.ToBig()) <= 0:
+			return fmt.Errorf("%w: new tx blob gas fee cap %v <= %v queued", txpool.ErrReplaceUnderpriced, tx.BlobGasFeeCap(), prev.blobFeeCap)
+		}
+		var (
+			multiplier = uint256.NewInt(100 + p.config.PriceBump)
+
+			minGasFeeCap     = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.execFeeCap), uint256.NewInt(100))
+			minGasTipCap     = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.execTipCap), uint256.NewInt(100))
+			minBlobGasFeeCap = new(uint256.Int).Div(new(uint256.Int).Mul(multiplier, prev.blobFeeCap), uint256.NewInt(100))
+		)
+		switch {
+		case tx.GasFeeCapIntCmp(minGasFeeCap.ToBig()) < 0:
+			return fmt.Errorf("%w: new tx gas fee cap %v <= %v queued + %d%% replacement penalty", txpool.ErrReplaceUnderpriced, tx.GasFeeCap(), prev.execFeeCap, p.config.PriceBump)
+		case tx.GasTipCapIntCmp(minGasTipCap.ToBig()) < 0:
+			return fmt.Errorf("%w: new tx gas tip cap %v <= %v queued + %d%% replacement penalty", txpool.ErrReplaceUnderpriced, tx.GasTipCap(), prev.execTipCap, p.config.PriceBump)
+		case tx.BlobGasFeeCapIntCmp(minBlobGasFeeCap.ToBig()) < 0:
+			return fmt.Errorf("%w: new tx blob gas fee cap %v <= %v queued + %d%% replacement penalty", txpool.ErrReplaceUnderpriced, tx.BlobGasFeeCap(), prev.blobFeeCap, p.config.PriceBump)
+		}
+	}
+	return nil
+}
+
+// Has returns an indicator whether subpool has a transaction cached with the
+// given hash.
+func (p *BlobPool) Has(hash common.Hash) bool {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	_, ok := p.lookup[hash]
+	return ok
+}
+
+// Get returns a transaction if it is contained in the pool, or nil otherwise.
+func (p *BlobPool) Get(hash common.Hash) *txpool.Transaction {
+	// Track the amount of time waiting to retrieve a fully resolved blob tx from
+	// the pool and the amount of time actually spent on pulling the data from disk.
+	getStart := time.Now()
+	p.lock.RLock()
+	getwaitHist.Update(time.Since(getStart).Nanoseconds())
+	defer p.lock.RUnlock()
+
+	defer func(start time.Time) {
+		gettimeHist.Update(time.Since(start).Nanoseconds())
+	}(time.Now())
+
+	// Pull the blob from disk and return an assembled response
+	id, ok := p.lookup[hash]
+	if !ok {
+		return nil
+	}
+	data, err := p.store.Get(id)
+	if err != nil {
+		log.Error("Tracked blob transaction missing from store", "hash", hash, "id", id, "err", err)
```

Comment:
Should we remove the hash from the lookup here? Or do you expect this to never happen and if it happens its a coding bug?

Reply:
Anything that's log.Error is a coding bug. So the more info the better.